packages/engine/src/worker/runner/python/batch.py

```diff
@@ -162,7 +159,7 @@ def flush_changes(self, schema, skip):
         # Dynamically accessed columns (if any) were added to `cols` by `state`.
         changes = []
         for field_name, col in self.cols.items():
-            if type(col) is not list or skip[field_name]:
+            if field_name in skip or len(col) == 0 or not isinstance(col, list):
```

Comment:
I don't really like it, but just saying that the Pythonic thing to do is say `or not col` rather than checking `len`

https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty

Reply:
Also do we want `not isinstance(col, list)` to be a check _before_ we check the `len`?

Reply:
I didn't know about `not col`. I would leave `len` if it's ok with you, because it seems clearer to me.

Reply:
I was thinking of `col` being either a list or an Arrow array, but you're right that checking the type first is safer.